build-scripts/esybuild.sh

```diff
@@ -0,0 +1,100 @@
+#!/bin/bash
+
+# =====================================
+# Globals
+# =====================================
+
+# shellcheck disable=SC2154
+readonly INSTALL_PREFIX=$cur__install
+readonly SOURCE_PATH=$(pwd)
+readonly BUILD_PATH="${SOURCE_PATH}/esy_build"
+
+# =====================================
+# Utils
+# =====================================
+
+# SDL2 build instructions specifies that we should use
+# a separate build directory instead of building in the
+# base of the source tree.
+# see: https://wiki.libsdl.org/Installation
+execute_inside_build_dir() {
+  [[ ! -d "$BUILD_PATH" ]] && mkdir "$BUILD_PATH"
+
+  (cd "$BUILD_PATH" && "$@")
+}
+
+make_and_install() {
+  make && make install
+}
+
+running_distro_is_arch() {
+  [[ -f "/etc/arch-release" ]]
+}
+
+# =====================================
+# Setup functions
+# =====================================
+
+setup_macos() {
+  "${SOURCE_PATH}"/configure \
+    --enable-shared=no \
+    --enable-static=yes \
+    --disable-video-x11 \
+    --prefix="$INSTALL_PREFIX"
+}
+
+setup_windows() {
+  SHELL=bash CC=x86_64-w64-mingw32-gcc WINDRES=x86_64-w64-mingw32-windres SDL_THREAD_PTHREAD=1 "${SOURCE_PATH}"/configure \
+    --prefix="$INSTALL_PREFIX" \
+    --enable-pthreads
+}
+
+setup_arch_linux() {
+  cmake "$SOURCE_PATH" \
+    -DCMAKE_INSTALL_PREFIX="$INSTALL_PREFIX" \
+    -DSDL_STATIC=OFF \
+    -DSDL_DLOPEN=ON \
+    -DARTS=OFF \
+    -DESD=OFF \
+    -DNAS=OFF \
+    -DALSA=ON \
+    -DPULSEAUDIO_SHARED=ON \
+    -DVIDEO_WAYLAND=OFF \
+    -DRPATH=OFF \
+    -DCLOCK_GETTIME=ON \
+    -DJACK_SHARED=ON
+}
+
+setup_linux_generic() {
```

Comment:
Moving all of the linux build to CMake seems reasonable to me. Thanks for preserving the existing behavior, though!